sdk/mediaservices/Microsoft.Azure.Media.LiveVideoAnalytics.Edge/src/MediaGraphPemCertificateList.cs

```diff
@@ -0,0 +1,95 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright company="Microsoft Corporation">
+//      Copyright (C) Microsoft Corporation. All rights reserved.
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Microsoft.Azure.Media.LiveVideoAnalytics.Edge.Models
+{
+    /// <summary>
+    /// Initializes a new instance of the MediaGraphPemCertificateList class.
+    /// </summary>
+    public partial class MediaGraphPemCertificateList
+    {
+        /// <summary>
+        /// How long the base 64 lines should be.
+        /// </summary>
+        private const int Base64LineLength = 76;
+
+        /// <summary>
+        /// Header for a PEM formatted certificate.
+        /// </summary>
+        /// <remarks>
+        /// See https://tools.ietf.org/html/rfc7468 Section 5.1.
+        /// </remarks>
+        private const string PemCertificateHeader = "-----BEGIN CERTIFICATE-----";
+
+        /// <summary>
+        /// Footer for a PEM formatted certificate.
+        /// </summary>
+        /// <remarks>
+        /// See https://tools.ietf.org/html/rfc7468 Section 5.1.
+        /// </remarks>
+        private const string PemCertificateFooter = "-----END CERTIFICATE-----";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MediaGraphPemCertificateList"/> class.
+        /// </summary>
+        /// <param name="certificates"> The list of certificates.</param>
+        public MediaGraphPemCertificateList(IList<X509Certificate2> certificates)
+        {
+            Certificates = certificates.Select(ConvertToPemString).ToList();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MediaGraphPemCertificateList"/> class.
+        /// </summary>
+        /// <param name="certificates"> The certificates params.</param>
+        public MediaGraphPemCertificateList(params X509Certificate2[] certificates)
+        {
+            Certificates = certificates.Select(ConvertToPemString).ToList();
+        }
+
+        private static string ConvertToPemString(X509Certificate2 certificate)
+        {
+            return new StringBuilder()
+                .AppendLine(PemCertificateHeader)
+
+                // Ideally this would be the following, but it is not supported in earlier versions of .NET
+                // .AppendLine(Convert.ToBase64String(certificate.Export(X509ContentType.Cert), Base64FormattingOptions.InsertLineBreaks))
+                .AppendLine(ToFormattedBase64String(certificate.Export(X509ContentType.Cert)))
+                .AppendLine(PemCertificateFooter)
+                .ToString();
+        }
+
+        private static string ToFormattedBase64String(byte[] data)
```

Comment:
Could you add a test for this?

Reply:
We don't have any tests for SDK right now, we plan to add tests by GA.

Reply:
We are dependent on tests on the Source repo, https://msazure.visualstudio.com/One/_git/Azure-Media-LiveVideoAnalytics?path=%2Fsrc%2FTest%2FComponentTests%2FEdge.Client%2FMediaGraphPemCertificateListTests.cs

This SDK is basically a copy of that from source repo,

Reply:
Ok. I would prefer to have tests for the code here, but if that is not feasible this would be fine.